app/models/spree/gateway/dolla_card.rb

```diff
@@ -0,0 +1,99 @@
+require "dolla"
+module Spree
+  class Gateway::DollaCard < Spree::PaymentMethod::CreditCard
+    TEST_VISA = ['4111111111111111', '4012888888881881', '4222222222222']
+    TEST_MC   = ['5500000000000004', '5555555555554444', '5105105105105100']
+    TEST_AMEX = ['378282246310005', '371449635398431', '378734493671000', '340000000000009']
+    TEST_DISC = ['6011000000000004', '6011111111111117', '6011000990139424']
+
+    VALID_CCS = ['1', TEST_VISA, TEST_MC, TEST_AMEX, TEST_DISC].flatten
+
+    attr_accessor :test
+
+    def gateway_class
+      self.class
+    end
+
+    def generate_profile_id(success)
+      record = true
+      prefix = success ? 'Dolla' : 'FAIL'
+      while record
+        random = "#{prefix}-#{Array.new(6){ rand(6) }.join}"
+        record = Spree::CreditCard.where(gateway_customer_profile_id: random).first
+      end
+      random
+    end
+
+    def create_profile(payment)
+      dolla_payment = Dolla::PaymentStub.new(
+        payment_id: payment.id,
+        code: payment.number,
+        amount: payment.amount.to_f,
+        cvv: payment.source.verification_value,
+        card_number: payment.source.number,
+        card_expiration: Date.parse("#{payment.source.year}/#{payment.source.month}"),
```

Comment:
this  needs to be a `Date` object, as you can see in https://github.com/amco/dolla/blob/master/lib/dolla/gateway.rb#L170 dolla will format it before sending the request

Reply:
oh, sorry it is `Date`